app/services/doc_auth/error_generator.rb

```diff
@@ -312,29 +328,51 @@ def generate_doc_auth_errors(response_info)
       metrics_error = metrics_error_handler.handle(response_info)
       return metrics_error.to_h if metrics_error.present? && !metrics_error.empty?
 
-      # check selfie error
-      selfie_error_handler = SelfieErrorHandler.new
-      selfie_error = selfie_error_handler.handle(response_info)
+      doc_auth_result = response_info[:doc_auth_result]
+      processed_alerts = response_info[:processed_alerts]
+
+      doc_auth_result_passed = doc_auth_result == LexisNexis::ResultCodes::PASSED.name
+      doc_auth_passed_or_attn_with_barcode = doc_auth_result_passed ||
+                                             attention_with_barcode_result(
+                                               doc_auth_result,
+                                               processed_alerts,
+                                             )
+      doc_auth_error_count =
+        doc_auth_passed_or_attn_with_barcode ?
+                               0 : response_info[:alert_failure_count]
+      known_error_count = doc_auth_error_count - unknown_fail_count
+
+      doc_auth_error_handler = DocAuthErrorHandler.new
+      doc_auth_error = doc_auth_error_handler.handle(response_info, known_error_count)
+
+      if doc_auth_error.present? && !doc_auth_error.empty?
+        return doc_auth_error.to_h
+      end
 
-      # if selfie itself is ok, but we have selfie related error
-      if selfie_error_handler.is_generic_selfie_error?(selfie_error)
-        return selfie_error_handler.selfie_general_failure_error
+      # check selfie error
+      if doc_auth_error_count < 1 && doc_auth_passed_or_attn_with_barcode
+        selfie_error_handler = SelfieErrorHandler.new
+        selfie_error = selfie_error_handler.handle(response_info)
+        if selfie_error.present? && !selfie_error.empty?
+          return selfie_error.to_h
+        end
       end
```

Comment:
I think this code can/should be restructured. Before this PR we had neat three line checks like this
```ruby
      # check Image metrics error
      metrics_error_handler = ImageMetricsErrorHandler.new(config)
      metrics_error = metrics_error_handler.handle(response_info)
      return metrics_error.to_h if metrics_error.present? && !metrics_error.empty?
```

It's probably not possible to do exactly that, but I think that all of this can at least be it's own method?
```ruby
      doc_auth_result = response_info[:doc_auth_result]
      processed_alerts = response_info[:processed_alerts]

      doc_auth_result_passed = doc_auth_result == LexisNexis::ResultCodes::PASSED.name
      doc_auth_passed_or_attn_with_barcode = doc_auth_result_passed ||
                                             attention_with_barcode_result(
                                               doc_auth_result,
                                               processed_alerts,
                                             )
      doc_auth_error_count =
        doc_auth_passed_or_attn_with_barcode ?
                               0 : response_info[:alert_failure_count]
```

Reply:
Great point! Fixed in [16068c9](https://github.com/18F/identity-idp/pull/10450/commits/16068c95e3226e4a785963d7555207c620c92d87).